src/api.cpp

```diff
@@ -287,6 +287,11 @@ void *jit_cuda_lookup(const char *name) {
     return jitc_cuda_lookup(name);
 }
 
+void jit_cuda_sync_stream(uint64_t stream) {
+    lock_guard guard(state.lock);
```

Comment:
No synchronization is needed for this operation, since you are only accessing the thread local state, but not the `state` variable. You can leave a comment saying so and remove the `lock_guard`.